web app/src/main/java/de/tinf15b4/kino/data/search/SearchService.java

```diff
@@ -20,22 +22,19 @@
     private CinemaRepository cinemaRepository;
 
     public SearchResult search(String searchTerm) {
-        List<Movie> movies = movieRepository.findByNameLike(searchTerm);
-        List<Cinema> cinemas = cinemaRepository.findNameLike(searchTerm);
-        List<Movie> movieDescriptions = movieRepository.findByDescriptionLike(searchTerm);
-        List<Cinema> cinemaAdresses = cinemaRepository.findByAddressLike(searchTerm);
+        Set<Cinema> cineSet = new HashSet<>();
+        Set<Movie> movieSet = new HashSet<>();
 
-        List<Movie> resultMovies = new ArrayList<>(movies.size() + movieDescriptions.size());
-        for (Movie m: movies)
-            resultMovies.add(m);
-        for (Movie m: movieDescriptions)
-            resultMovies.add(m);
+        movieSet.addAll(movieRepository.findByNameLike(searchTerm));
+        cineSet.addAll(cinemaRepository.findNameLike(searchTerm));
+        movieSet.addAll(movieRepository.findByDescriptionLike(searchTerm));
+        cineSet.addAll(cinemaRepository.findByAddressLike(searchTerm));
 
         List<Cinema> resultCinemas = new ArrayList<>();
-        for (Cinema c : cinemas)
-            resultCinemas.add(c);
-        for (Cinema c : cinemaAdresses)
-            resultCinemas.add(c);
+        List<Movie> resultMovies = new ArrayList<>();
+
+        resultCinemas.addAll(cineSet);
+        resultMovies.addAll(movieSet);
```

Comment:
Might be a little shorter and nicer looking: 
List<Movie> resultMovies = Lists.newArrayList<>(movies);
List<Cinema> resultCinemas = Lists.newArrayList<>(cinemas);